core/src/main/scala/filodb.core/memstore/TimeSeriesMemStore.scala

```diff
@@ -135,7 +145,9 @@ extends MemStore with StrictLogging {
                   }.collect { case Some(flushGroup) => flushGroup }
                   .mapAsync(numParallelFlushes) { f => shard.createFlushTask(f).executeOn(flushSched).asyncBoundary }
                            // asyncBoundary so subsequent computations in pipeline happen in default threadpool
-                  .foreach({ x => })(shard.ingestSched)
+                  .completedL
+                  .doOnCancel(cancelTask)
```

Comment:
We also need to add a doOnFinish handler with same cleanup tasks right ?

Reply:
On normal completion or failure, the caller can continue doing that on the future object that's returned.

Reply:
I recommend passing single onEnd handler:

```
def onEnd(ex: Option[Throwable]): Task[Unit] = {
    ex match {
       case None => val stopped = IngestionStopped(dataset.ref, shard)
                                self ! stopped
                                statusActor ! stopped
       case Some(ex) => 
                                self ! IngestionStopped(dataset.ref, shard)
                                statusActor ! IngestionError(dataset.ref, shard)
}
```

and attach
```
.doOnCancel(Task { onEnd(None) } )
.doOnFinish(ex: Task { onEnd(ex) } )
```

Granted, there is a bit more cleanup to be done to the above code, but you get the idea ...